server/env.ts

```diff
@@ -121,41 +129,49 @@ export class Environment {
   /**
    * The fully qualified, external facing domain name of the server.
    */
+  @Public
   @IsNotEmpty()
   @IsUrl({
     protocols: ["http", "https"],
     require_protocol: true,
     require_tld: false,
   })
-  public URL = environment.URL || "";
+  public URL = (environment.URL ?? "").replace(/\/$/, "");
 
   /**
    * If using a Cloudfront/Cloudflare distribution or similar it can be set below.
    * This will cause paths to javascript, stylesheets, and images to be updated to
    * the hostname defined in CDN_URL. In your CDN configuration the origin server
    * should be set to the same as URL.
    */
+  @Public
   @IsOptional()
   @IsUrl({
     protocols: ["http", "https"],
     require_protocol: true,
     require_tld: false,
   })
-  public CDN_URL = this.toOptionalString(environment.CDN_URL);
+  public CDN_URL = this.toOptionalString(
+    environment.CDN_URL ? environment.CDN_URL.replace(/\/$/, "") : ""
```

Comment:
```suggestion
    environment.CDN_URL ? environment.CDN_URL.replace(/\/$/, "") : undefined
```

Given it's optional `undefined` feels more correct?